Content.Server/Store/Systems/StoreSystem.Ui.cs

```diff
@@ -225,4 +306,71 @@ private void OnRequestWithdraw(EntityUid uid, StoreComponent component, StoreReq
         component.Balance[msg.Currency] -= msg.Amount;
         UpdateUserInterface(buyer, uid, component);
     }
+
+    private void OnRequestRefund(EntityUid uid, StoreComponent component, StoreRequestRefundMessage args)
+    {
+        // TODO: Remove guardian/holopara
+
+        if (args.Session.AttachedEntity is not { Valid: true } buyer)
+            return;
+
+        if (!IsOnStartingMap(uid, component))
+        {
+            component.RefundAllowed = false;
+            UpdateUserInterface(buyer, uid, component);
+        }
+
+        if (!component.RefundAllowed || component.BoughtEntities.Count == 0)
+            return;
+
+        for (var i = component.BoughtEntities.Count; i >= 0; i--)
```

Comment:
Either I'm stupid or this one is surely supposed to crash on clean fork cause component.BoughtEntities[i] will throw OOR.. Make it `var i = component.BoughtEntities.Count - 1`?